quickjs.c

```diff
@@ -47785,11 +47787,13 @@ static BOOL js_date_parse_otherstring(const uint8_t *sp,
             string_skip_until(sp, &p, "0123456789 -/(");
         } else
         if (has_time && string_match(sp, &p, "PM")) {
-            if (fields[3] < 12)
+            if (fields[3] != 12)
```

Comment:
Should this also have a range check?
```suggestion
            if (fields[3] > 12)
                return FALSE;
            if (fields[3] != 12)
```
It looks like it'd accept e.g. `13:00 PM` now.

Reply:
The test `fields[3] != 12` suffices to reject hours larger than 12 because they will be incremented and exceed 24, thus be caught by the range check in `js_Date_parse()`.

Reply:
I shall add an explanatory comment for clarity